typed-builder-macro/src/struct_info.rs

```diff
@@ -680,88 +680,70 @@ impl<'a> TypeBuilderAttr<'a> {
         Ok(result)
     }
 
-    fn apply_meta(&mut self, expr: syn::Expr) -> Result<(), Error> {
+    fn apply_meta(&mut self, expr: AttrArg) -> Result<(), Error> {
         match expr {
-            syn::Expr::Assign(assign) => {
-                let name =
-                    expr_to_single_string(&assign.left).ok_or_else(|| Error::new_spanned(&assign.left, "Expected identifier"))?;
-
+            AttrArg::KeyValue(ref kv @ KeyValue { ref name, ref value, .. }) => {
```

Comment:
I'm not sure, if the better approach might be to use `kv.name` and `kv.value` instead.
```suggestion
            AttrArg::KeyValue(kv) => {
```

Reply:
Do you even _need_ `kv`? You are only using it in this line:

```rust
_ => Err(Error::new_spanned(kv, format!("Unknown parameter {:?}", name))),
```

But couldn't you just pass `expr`? It has the same tokens as `kv`.

Reply:
korrect, but I'd still need to do `ref` on `name` and `value` anyway.

Reply:
Yes, you'd need that anyway, but that's a matter of style.